fms/modules/head.py

```diff
@@ -1,10 +1,19 @@
-from typing import Optional
+from typing import Dict, List, Optional, Set
 
 import torch
+import torch.distributed
 import torch.nn as nn
+from torch.distributed.distributed_c10d import ProcessGroup
 
+from fms import distributed
+from fms.distributed.tensorparallel import (
+    all_gather_from_tensor_model_parallel_region,
+    copy_to_tensor_model_parallel_region,
+)
+from fms.modules.tp import TPModule
 
-class ClassificationHead(nn.Module):
+
+class MLPClassificationHead(nn.Module):
```

Comment:
and is this used? will whatever model uses it require a TP variant too?

Reply:
This is Roberta's head, so it is used. As far as TP goes, I don't think a Roberta variant exists that requires TP in any of our workloads. If/when it's needed, we can add it.